specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-12-01-preview/containerregistry.json

```diff
@@ -0,0 +1,1291 @@
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerRegistry/registries/{registryName}/listCredentials": {
+      "post": {
+        "tags": [
+          "Registries"
+        ],
+        "description": "Lists the login credentials for the specified container registry.",
+        "operationId": "Registries_ListCredentials",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/RegistryNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request was successful; the list of credentials retrieved and returned successfully.",
+            "schema": {
+              "$ref": "#/definitions/RegistryListCredentialsResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "RegistryListCredentials": {
+            "$ref": "./examples/RegistryListCredentials.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerRegistry/registries/{registryName}/regenerateCredential": {
+      "post": {
+        "tags": [
+          "Registries"
+        ],
+        "description": "Regenerates one of the login credentials for the specified container registry.",
+        "operationId": "Registries_RegenerateCredential",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/RegistryNameParameter"
+          },
+          {
+            "name": "regenerateCredentialParameters",
+            "in": "body",
+            "description": "Specifies name of the password which should be regenerated -- password or password2.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RegenerateCredentialParameters"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request was successful; the specified credential regenerated successfully.",
+            "schema": {
+              "$ref": "#/definitions/RegistryListCredentialsResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "RegistryRegenerateCredential": {
+            "$ref": "./examples/RegistryRegenerateCredential.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerRegistry/registries/{registryName}/listUsages": {
```

Comment:
Why is this "listUsages" and not /usages? It should follow the[ contract](https://armwiki.azurewebsites.net/api_contracts/UsagesAPIContract.html).

Reply:
I do not want to make changes to the old controller as part of this PR. I will open WI to track these changes.

Reply:
Can you help me understand why a new version (a preview too) is not a good opportunity to correct issues? I'm not sure what you mean by "old controller/api"

Reply:
changes the api and made it usages from listUsages

Reply:
@KrisBash  I see that there is a contract to return the resource usages across Azure. But this API is not exactly the contract implementation as the payload is different and back then we didn't have this contract. Also, it loses its ability to call from ARM template if we remove **list** prefix. We need to evaluate our API usage and then introduce this change. Can we please take this change as a separate item and scope this version to customer managed key scenarios.